app/models/invenio_rdm_record_converter.rb

```diff
@@ -227,54 +227,47 @@ def build_creator_contributor_json(creator)
       given_name = given_name.gsub(',', "")
     end
 
-    if dh_user = User.find_by(formal_name: creator) || User.find_by(display_name: display_name)
-      identifiers = dh_user.orcid.present? ? [{"scheme": "orcid", "identifier": dh_user.orcid.split('/').pop}] : nil
-    end
+    dh_user = User.find_by(formal_name: creator) || User.find_by(display_name: display_name)
 
     # Known organization
     if organization?(creator)
-      json = {
+      creatibutor_json = {
         "person_or_org": {
           "name": creator,
           "type": "organizational"
         }
       }
-    # Personal record with user in database
-    elsif dh_user.present?
-      json = {
+    # Personal record with user in database OR is a person's formal name
+    elsif dh_user.present? || creator.include?(",")
+      creatibutor_json = {
         "person_or_org": {
-          "given_name": given_name,
-          "family_name": family_name,
-          "type": "personal",
-          "identifiers": identifiers
-        }
-      }
-    # Personal record without user in database
-    elsif creator.include?(",")
-      json = {
-        "person_or_org":
-        {
           "given_name": given_name,
           "family_name": family_name,
           "type": "personal"
         }
       }
+
+      if dh_user.orcid.present?
+        identifiers = [{"scheme": "orcid", "identifier": dh_user.orcid.split('/').pop}]
+        creatibutor_json["person_or_org"].merge!({"identifiers": identifiers})
+      end
+    # Personal record without user in database
     # Unknown / Not Identified creator
```

Comment:
I'll remove this comment quickly

Reply:
Eh, I'm just going to leave it. I'd have to add the commit outside of release or we'd end up with more annoying merge conflicts in the future.